### PR TITLE
Custom http service

### PR DESCRIPTION
**Pull request by GuillaumeNury**
Allow both `luid-user-picker` and `luid-user-picker-multiple` to handle cutomHttpService

### CODE REVIEW

**Comment by jbbeuzelin** [comment]
:+1:

File: js/directives/lucca/user-picker.js [hunk 1]
```diff
@@ -188,6 +192,14 @@
...
 		var myId; // used for 'display me first' feature
 
+		/** HttpService **/
+		var httpService = function(method){
+			if($scope.customHttpService &&  $scope.customHttpService[method]){
+				return $scope.customHttpService[method];
+			}
+			return $http[method];
+		};
+
```
Comment (lucienbertin) [inline js/directives/lucca/user-picker.js]: I don't like the fact that your function is called `httpService` but returns a `httpMethod`

File: tests/spec/directives/user-picker.spec.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-describe('luidUserPicker', function(){
 	beforeEach(module('moment'));
+	describe('luidUserPicker', function(){
```
Comment (lucienbertin) [inline tests/spec/directives/user-picker.spec.js]: ??? why this diff ?

File: tests/spec/directives/user-picker.spec.js [hunk 3]
```diff
@@ -1009,6 +1009,43 @@ describe('luidUserPicker', function(){
...
+		var meApi = /api\/v3\/users\/me/;
+		it('should call the given "get" method', function() {
+			var tpl = angular.element('<luid-user-picker ng-model="chloe" custom-http-service="customHttpService"></luid-user-picker>');
+			$scope.customHttpService = customHttpService;
+			elt = $compile(tpl)($scope);
+			isolateScope = elt.isolateScope();
+			spyOn($scope.customHttpService, 'get').and.callThrough();
+			isolateScope.find();
+			$scope.$digest();
+			expect($scope.customHttpService.get).toHaveBeenCalled();
```
Comment (lucienbertin) [inline tests/spec/directives/user-picker.spec.js]: you're not testing `$http.get` isnt called anymore

File: tests/spec/directives/user-picker.spec.js [hunk 4]
```diff
@@ -1009,6 +1009,43 @@ describe('luidUserPicker', function(){
...
+	describe("with customHttpService", function(){
+		var chloe = { id:3,
+			firstName:"Chloé",
+			lastName:"Azibert Yekdah"
+		};
+		var customHttpService = {
+			get: function(query){
+				return $q.defer().promise;
+			}
+		};
```
Comment (lucienbertin) [inline tests/spec/directives/user-picker.spec.js]: there are other calls to the get method done after the first get resolves, when there are homonyms. your tests does not cover that in that case `$http.get` isnt called

**Pull request merged**